src/test/java/commission/services/PaymentCalculatorServiceTest.java

```diff
@@ -0,0 +1,202 @@
+import commission.service.PaymentCalculatorService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+
+import static commission.entity.CommissionType.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class PaymentCalculatorServiceTest {
+
+    @Mock
+    private SaleDaoImpl saleDao;
+    @Mock
+    private CommissionDaoImpl commissionDao;
+
+    @InjectMocks
+    private PaymentCalculatorService paymentCalculatorService;
+
+    @BeforeEach
+    public void setUp(){
+        paymentCalculatorService = new PaymentCalculatorService(saleDao, commissionDao);
+
+        when(saleDao.findById(1)).thenReturn(Optional.empty());
+        when(saleDao.findById(2)).thenReturn(Optional.of(new Sale(2, 2000, OffsetDateTime.parse("2024-01-10T10:15:30+02:00") , 12)));
+        when(saleDao.findById(3)).thenReturn(Optional.of(new Sale(3, 3000, OffsetDateTime.parse("2024-02-03T10:15:30+02:00"), 12)));
+        when(saleDao.findById(4)).thenReturn(Optional.of(new Sale(4, 1000, OffsetDateTime.parse("2024-03-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(5)).thenReturn(Optional.of(new Sale(5, 7000, OffsetDateTime.parse("2024-03-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(6)).thenReturn(Optional.of(new Sale(6, 14000, OffsetDateTime.parse("2024-04-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(7)).thenReturn(Optional.of(new Sale(7, 5000, OffsetDateTime.parse("2024-05-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(8)).thenReturn(Optional.of(new Sale(8, 10000, OffsetDateTime.parse("2024-06-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(10)).thenReturn(Optional.of(new Sale(10, 0, OffsetDateTime.parse("2024-12-10T10:15:30+02:00") , 20)));
+
+        when(commissionDao.findById(1)).thenReturn(Optional.empty());
+        when(commissionDao.findById(2)).thenReturn(Optional.of(new Commission(2, STRAIGHT, 200, 2)));
+        when(commissionDao.findById(3)).thenReturn(Optional.of(new Commission(3, GROSS_MARGIN, 120, 3)));
+        when(commissionDao.findById(4)).thenReturn(Optional.of(new Commission(4, TIERED, 100, 4)));
+        when(commissionDao.findById(5)).thenReturn(Optional.of(new Commission(5, TIERED, 1050, 5)));
+        when(commissionDao.findById(6)).thenReturn(Optional.of(new Commission(6, TIERED, 2800, 6)));
+        when(commissionDao.findById(7)).thenReturn(Optional.of(new Commission(7, TIERED, 750, 7)));
+        when(commissionDao.findById(8)).thenReturn(Optional.of(new Commission(8, TIERED, 1500, 8)));
+        when(commissionDao.findById(10)).thenReturn(Optional.of(new Commission(10, TIERED, 1, 10)));
```

Comment:
You don't need to use a mock for this, if you want to use the same commission instance in multiple tests you can either:
1. Define a variable at the class level for the commission
2. Duplicate the code in the tests. This is okay for tests because you want each test to have their own version of test data. Tests should be self encapsulating as much as possible.

Mocks shouldn't be used to retrieve data in your tests method. They are used to pass on to other classes to returned "mocked" information when code interacts with the mock

Reply:
That makes more sense. I did the same with saleDao mock. Is specific to each test too.

Reply:
There isn't a black / white rule on this, sometimes it's nicer to have the test data for mocks in the tests itself and sometimes you can define them more globally. It is more important for the input and expected output data to be defined in the test so it makes it easier to reason about. This is all for the sake of maintenance